### PR TITLE
Enhance job run status

### PR DESCRIPTION
**Pull request by yhwen**
Added update_job_run_status to update the proper job run status.

Description:
At the end of ServerAppRunner application job run, it will check if the job run has FLContextKey.FATAL_SYSTEM_ERROR been reported. If it does, then post the job execution exception status to the parent process to update the job FINISHED:FINISHED_EXECUTION_EXCEPTION outcome. Otherwise, the job outcome will be FINISHED:COMPLETED.

system panic and server job exception will mark as FLContextKey.FATAL_SYSTEM_ERROR.

### CODE REVIEW

File: nvflare/private/fed/server/server_engine.py [hunk 1]
```diff
@@ -254,11 +266,7 @@ def wait_for_complete(self, job_id):
             except BaseException:
                 with self.lock:
                     if job_id in self.run_processes:
-                        run_process_info = self.run_processes.pop(job_id)
-                        return_code = run_process_info[RunProcessKey.CHILD_PROCESS].poll()
-                        # if process exit but with Execution exception
-                        if return_code and return_code != 0:
-                            self.execution_exception_run_processes[job_id] = run_process_info
+                        self.run_processes.pop(job_id)
                 self.engine_info.status = MachineStatus.STOPPED
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_engine.py:270]: The original logic covers the situation where an exception happens in the "finally" block in `nvflare/private/fed/server/server_app_runner.py`

Also, runner_process will throw exception as well: `https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/private/fed/app/server/runner_process.py#L129`

This new logic does not account for those situations.

Comment (yhwen) [reply nvflare/private/fed/server/server_engine.py:270]: This already covered in ServerAppRunner.start_server_app() finally block.  No need for the old logic any more.

File: nvflare/private/fed/server/job_runner.py [hunk 2]
```diff
@@ -301,6 +301,7 @@ def _job_complete_process(self, fl_ctx: FLContext):
                             fl_ctx.set_prop(FLContextKey.CURRENT_JOB_ID, job.job_id)
                             self.fire_event(EventType.JOB_COMPLETED, fl_ctx)
                             self.log_debug(fl_ctx, f"Finished running job:{job.job_id}")
+                    engine.remove_exception_process(job_id)
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/server/job_runner.py:304]: Once job outcome has been updated, no need to keep the job's execution error info any more.

File: nvflare/private/fed/server/server_engine.py (hunk 1 above)
Comment (yhwen) [reply nvflare/private/fed/server/server_engine.py:274]: rolled back the wait_for_complete logic.

**Pull request merged**